### ISSUE TITLE
Links aren't properly rendered when preceded by quoted text

### ISSUE THREAD

**Issue by gsrohde** [doxygen/doxygen#8525]
**Describe the bug**
When some double-quoted text that extends over three or more lines is immediately followed by a link, the link isn't rendered properly.

**Expected behavior**
A link of the form `[link text](linkurl.com)` should generate HTML of the form `<a href="linkurl.com">link text</a>`.  Instead I am seeing `&lt;a href="linkurl.com"&gt;link text`, that is, the opening HTML tag delimiters are escaped, and the closing tag is missing.

**To Reproduce**
Make a Doxyfile with this content:
```
# Doxyfile 1.9.1

EXTRACT_ALL            = YES
INPUT                  = my_function.cpp
```
and a source file `my_function.cpp` in the same directory with this content:
```
/**
 *  \brief my_function returns a double.
 *
 *  References
 *
 *  "A two line quotation followed by
 *   a reference"
 *   [See an example](example.com)
 *
 *  "A three line quotation
 *   followed by
 *   a reference"
 *   [See an example](example.com)
 *
 *  "A three line quotation
 *   followed by a line of text and then
 *   a reference"
 *   This works.
 *   [See an example](example.com)
 *
 *  The same references, but using reference links rather than inline
 *  links:
 *
 * [link name]: example.com
 *
 *
 *  "A two line quotation followed by
 *   a reference"
 *   [See an example][link name]
 *
 *  "A three line quotation
 *   followed by
 *   a reference"
 *   [See an example][link name]
 *
 *  "A three line quotation
 *   followed by a line of text and then
 *   a reference"
 *   This works.
 *   [See an example][link name]
 *
 */
double my_function() {
    return 1.234;
}
```

Run `doxygen` in the directory containing these files and open the generated HTML file (`my__function_8cpp.html` on my system).

The portion of the generated HTML corresponding to the references looks like this:
```
<p>my_function returns a double. </p>
<p>References</p>
<p>"A two line quotation followed by
  a reference" <a href="example.com">See an example</a></p>
<p>"A three line quotation followed by a reference"
  &lt;a href="example.com"&gt;See an example</p>
<p>"A three line quotation followed by a line of text and then a reference" This works. <a href="example.com">See an example</a></p>
<p>The same references, but using reference links rather than inline links:</p>
<p>"A two line quotation followed by
  a reference" <a href="example.com">See an example</a></p>
<p>"A three line quotation followed by a reference"
  &lt;a href="example.com"&gt;See an example</p>
<p>"A three line quotation followed by a line of text and then a reference" This works. <a href="example.com">See an example</a> </p>
```
The generated PDF file suffers from similar problems.

**Version**
I used Doxygen version 1.9.1 on a Mac (macOS 10.14.6); it was installed from the DMG file.

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #8527

**Comment by albert-github** [comment]
Code has been integrated in master on github (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.2.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).

**Comment by gsrohde** [comment]
Yes, this seems to be fixed in version 1.9.2, at least for the HTML docs.  (I had trouble building the LaTeX PDF—probably due to a problem with my LaTeX installation and not anything in this release.)

I used Doxygen version 1.9.2 on a Mac (macOS 10.15.7); it was installed from the DMG file.
